### PR TITLE
[Sealing Segment] Handle root sealing segment with multiple blocks

### PR DESCRIPTION
**Pull request by jordanschalm**
This PR adds a test case and extends the implementation of `SealingSegment` to deal with an unhandled edge case:
* sealing segment contains the root block
* sealing segment contains more than one block 

This edge case shows up in integration tests, when we request a root snapshot immediately after the network starts up.

### CODE REVIEW

File: model/flow/sealing_segment.go [hunk 1]
```diff
@@ -30,27 +20,41 @@ const (
...
+// Example 1 - one self-sealing root block
+//   ROOT
+// The above sealing segment is the form of sealing segments within root snapshots,
+// for example those snapshots used to bootstrap a new network, or spork.
+//
+// Example 2 - one self-sealing root block followed by any number of seal-less blocks
+//   ROOT <- A <- B
+//
+// All non-root sealing segments contain more than one block.
+// Sealing segments is in ascending height order.
```
Comment (kc1116) [inline model/flow/sealing_segment.go:47]:
```suggestion
// Sealing segments are in ascending height order.
```

File: model/flow/sealing_segment.go [hunk 2]
```diff
@@ -30,27 +20,41 @@ const (
...
-// - results referenced by receipts within segment payloads
-// - results referenced by seals within segment payloads
-// - seals which represent the latest state commitment as of a segment block
-//
+// * results referenced by receipts within segment payloads
+// * results referenced by seals within segment payloads
+// * seals which represent the latest state commitment as of a segment block
 type SealingSegment struct {
-	// Blocks contains the chain segment blocks.
+	// Blocks contains the chain segment blocks in ascending height order.
```
Comment (kc1116) [inline model/flow/sealing_segment.go:57]:
```suggestion
	// Blocks contain the chain segment blocks in ascending height order.
```

File: model/flow/sealing_segment.go [hunk 3]
```diff
@@ -30,27 +20,41 @@ const (
...
 	Blocks []*Block
 
 	// ExecutionResults contains any results which are referenced by receipts
```
Comment (kc1116) [inline model/flow/sealing_segment.go:60]:
```suggestion
	// ExecutionResults contain any results which are referenced by receipts
```

File: model/flow/sealing_segment.go [hunk 4]
```diff
@@ -87,10 +91,10 @@ func (segment *SealingSegment) Lowest() *Block {
 
 // FinalizedSeal returns the seal that seals the lowest block.
 // Per specification, this seal must be included in a SealingSegment.
-// Under normal operations (where only a validated SealingSegment
-// is processed), no error returns are expected.
+// The receiver SealingSegment must be validated.
```
Comment (kc1116) [inline model/flow/sealing_segment.go:94]: What is meant by "receiver sealing segment " ?

**Review by kc1116 — APPROVED** [review]

File: model/flow/sealing_segment.go (hunk 4 above)
Comment (jordanschalm) [reply model/flow/sealing_segment.go:94]: The receiver is the name for the parameter specified in the first set of braces.
```go
func (r Receiver) DoSomething(...)
```

Comment (jordanschalm) [inline model/flow/sealing_segment.go:94]:
```suggestion
// The SealingSegment must be validated.
```

Comment (kc1116) [reply model/flow/sealing_segment.go]: Gotcha , "The receiver" sounds like it is something related to the sealing segment. +1 for removing it.

**Review by Kay-Zee — APPROVED** [review]

**Review by durkmurder — APPROVED** [review]
:guitar:

File: model/flow/sealing_segment.go [hunk 5]
```diff
@@ -30,30 +20,44 @@ const (
...
 // _not_ yield a valid SealingSegment.
 //
-// Root sealing segments contain only one self-sealing block. All other sealing
-// segments contain multiple blocks.
-// The segment is in ascending height order.
+// ROOT SEALING SEGMENTS:
+// Root sealing segments are sealing segments which contain the root block:
+// * the root block must be the first block (least height) in the segment
+// * no blocks in the segment may contain any seals (by the minimality requirement)
+// * it is possible (but not necessary) for root sealing segments to contain only 1 self-sealing block
```
[the remark this answers is not in the document]
Comment (jordanschalm) [reply model/flow/sealing_segment.go]: https://github.com/onflow/flow-go/pull/2786/commits/bca3b55e28d4487ecd5f0618e8ef4ba110f69a6d
